### ISSUE TITLE
[Feature Request] Option for disable esm/cjs shim

### ISSUE THREAD

**Issue by Amorites** [egoist/tsup#485]
I'm currently using `tsup` to pack some packages used in a WeChat mini program that doesn't have a `URL`.

https://github.com/egoist/tsup/commit/d0870dd3bdf939d03d577355de8f56583c8be403 introduce `esm_shim.js` in esm build.

And somehow `cjs_shim.js` is introduced in cjs build (still I haven't figured out why)

**Comment by egoist** [comment]
work around:

```ts
export default defineConfig({
  esbuildOptions(opts) {
    opts.inject = opts.inject.filter(p => !p.includes('_shims'))
  }
})
```

**Comment by egoist** [comment]
btw can you provide a repo?

**Comment by Amorites** [comment]
> btw can you provide a repro?

https://github.com/Amorites/tsup-cjs-shim-issue

Here is a repro, it does not recur in non-monorepo modes

**Comment by egoist** [comment]
this should be fixed in 5.11.10, let me know if it still doesn't work
